src/components/tests/SelectedRoute.test.tsx

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import { render } from "@testing-library/react";
+
+import SelectedRoute from "../SelectedRoute";
+import { MRT_BLUE_STATION, LineType, BTS_SUKHUMVIT_STATION, TravelRoute, Station, BTS_SILOM_STATION } from "../../types";
+
+let travelRoute: TravelRoute;
+let station: Station;
+
+import * as uiservice from "../../services/ui-style.service";
+jest.mock("../../services/ui-style.service");
+
+const mockGetInterChangeLine = uiservice.getInterChangeLine as jest.MockedFunction<typeof uiservice.getInterChangeLine>;
+const mockGetDottedLineStyle = uiservice.getDottedLineStyle as jest.MockedFunction<typeof uiservice.getDottedLineStyle>;
+const mockGetStationIconStyle = uiservice.getStationIconStyle as jest.MockedFunction<typeof uiservice.getStationIconStyle>;
+
+import * as service from "../../services/util.service";
+jest.mock("../../services/util.service");
+
+const mockGetStation = service.getStation as jest.MockedFunction<typeof service.getStation>;
+const mockGetStationName = service.getStationName as jest.MockedFunction<typeof service.getStationName>;
+
+let mockedTranslate = jest.fn();
+let mockedChangeLanguage = jest.fn();
+jest.mock("react-i18next", () => {
+  return {
+    useTranslation: () => {
+      return ({
+        i18n: {
+          changeLanguage: mockedChangeLanguage,
+        },
+        t: mockedTranslate,
+      })
+    }
+  }
+});
+
+beforeEach(() => {
+  jest.clearAllMocks();
+  station = {
+    lineType: LineType.MRT_BLUE,
+    key: MRT_BLUE_STATION.BANG_WA,
+    nameEN: 'english',
+    nameTH: 'thai',
+    isNotAvailable: false,
+    position: [0, 0]
+  };
+  travelRoute = {
+    route: [
+      {
+        route: [MRT_BLUE_STATION.SILOM, MRT_BLUE_STATION.LUMPHINI],
+        lineType: LineType.MRT_BLUE,
+        fare: 0
+      },
+      {
+        route: [BTS_SUKHUMVIT_STATION.ARI, BTS_SUKHUMVIT_STATION.ASOK],
+        lineType: LineType.BTS_SUKHUMVIT,
+        fare: 0
+      }
+    ],
+    fare: 0,
+    source: MRT_BLUE_STATION.SILOM,
+    destination: MRT_BLUE_STATION.LUMPHINI
+  };
+  mockGetStation.mockReturnValue(station);
+});
+
+describe("<SelectedRoute />", () => {
+  test("should render the component without errors", async () => {
+    render(<SelectedRoute travelRoute={travelRoute}/>);;
+  });
+
+  test("should show the fare and translated text", async () => {
+    mockedTranslate.mockReturnValue("Fare");
+    travelRoute.fare = 150;
```

Comment:
I think it will be better if we don't mutate `travelRoute`. I normally do and prefer this way.
```
const newRoute = {
    ...travelRoute,
    fare: 150
}
```